### PR TITLE
[6.4] macOS 27.0 architecture updates

### PR DESCRIPTION
**Pull request by owenv**
macOS 27.0 no longer supports x86_64. Mark the architecture deprecated so we only build an x86 slice by default when using a deployment target macOS 26 or earlier. This PR also updates a number of tests to account for this change.

### CODE REVIEW

**Comment by owenv** [comment]
Starting as a draft while I confirm all the test updates are correct

**Comment by owenv** [comment]
@swift-ci test

File: Sources/SWBApplePlatform/Specs/DriverKit.xcspec [hunk 1]
```diff
@@ -82,6 +82,7 @@
         Name = "Intel 64-bit";
         Description = "64-bit Intel";
         PerArchBuildSettingName = "Intel 64-bit";
+        DeploymentTargetRange = ( "0", "27" );
```
Comment (jakepetroules) [inline Sources/SWBApplePlatform/Specs/DriverKit.xcspec:85]: Super pedantic but... since DriverKit 26 doesn't exist and is aliased to DriverKit 27, should the upper bound be set to 26 so that erroneously specified deployment targets don't try to build for x86_64?

"that's enough of an edge case that we can ignore it" is also an acceptable answer.

**Review by jakepetroules — APPROVED** [review]

**Review by mhrawdon — APPROVED** [review]

**Comment by owenv** [comment]
@swift-ci test

File: Sources/SWBApplePlatform/Specs/DriverKit.xcspec (hunk 1 above)
Comment (owenv) [reply Sources/SWBApplePlatform/Specs/DriverKit.xcspec:85]: That feels a bit too subtle for me, I think I'd almost prefer diagnosing any attempt to use the nonexistent version

Comment (jakepetroules) [reply Sources/SWBApplePlatform/Specs/DriverKit.xcspec:85]: Agree. Just wanted to call it out.

**Comment by owenv** [comment]
@swift-ci test

**Comment by owenv** [comment]
@swift-ci test macOS

**Comment by owenv** [comment]
@swift-ci test linux

**Comment by owenv** [comment]
@swift-ci test

**Comment by owenv** [comment]
@swift-ci test windows

**Comment by owenv** [comment]
@swift-ci test

**Comment by owenv** [comment]
@swift-ci test macOS

**Comment by owenv** [comment]
@swift-ci test linux
